### DISCUSSION TITLE
Getting 403 while connecting to websocket

### DISCUSSION THREAD

**Discussion by vasu-dholakiya** [miguelgrinberg/python-socketio#1247] [Q&A]
I have used python-socketio with FastAPI & Uvicorn as server, But when I try to connect to socket using the below mentioned URL getting 403 as response 

Dependency:
python-engineio    3.14.2
python-socketio    4.6.1
uvicorn            0.22.0
fastapi            0.97.0

URL to connect: ws://localhost:5050/ws

**Code:**

```
import uvicorn
from engineio import ASGIApp
from fastapi import FastAPI
from socketio import AsyncServer

app = FastAPI(title=__name__)

sio = AsyncServer(async_mode="asgi", cors_allowed_origins="*", always_connect=True)
socket_app = ASGIApp(sio, other_asgi_app=app)
app.mount(path='/ws', app=socket_app, name="SocketServer")

if __name__ == "__main__":
    uvicorn.run(
        "__main__:app",
        host="0.0.0.0",
        port=5050,
        reload=False
    )

```

***Response***

Error: Unexpected server response: 403

**Comment by miguelgrinberg** [comment]
Please follow the rules of this repository. Do not write issues when you need help solving a problem on your own application. The issues board is for bugs that need to be fixed on this package. Use the discussions board to ask questions or troubleshoot an issue on your own application. Thank you.

Basically your problem is that you are mixing up the FastAPI and the Socket.IO ASGI applications in a strange way. You have to decide if you want to use the `ASGIApp` solution provided by this package to create a combined ASGI app, or if you want to use FastAPI's mount solution. It looks like you have a mix of both in your code.

This package tries to be agnostic to what other libraries or frameworks you use, so the FastAPI specific solution is not something I intend to support (though I'm sure it is possible to make it work). For the `ASGIApp` solution, I have added an [example](https://github.com/miguelgrinberg/python-socketio/blob/main/examples/server/asgi/fastapi-fiddle.py) to this repository that hopefully makes it more clear. I would recommend that you do not change the standard path of the Socket.IO endpoint, but if you must change it, you can do so with the `socketio_path` argument to the `ASGIApp` constructor.

**Reply by vasu-dholakiya** [reply]
Thanks for the explanation, Btw getting same 403 in the provided [example](https://github.com/miguelgrinberg/python-socketio/blob/main/examples/server/asgi/fastapi-fiddle.py) , If you can help me to resolve this out!!

**Reply by miguelgrinberg** [reply]
The example works correctly. My guess is that your client is either not a Socket.IO client, or if it is, it is connecting to the wrong URL to get the Socket.IO connection. Use your browser to connect to `http://localhost:5000` with the example code I uploaded to see a working solution.

**Reply by vasu-dholakiya** [reply]
I have used Postman as well as this [tool ](https://websocketking.com/), With the below URL to connect a socket, And it's giving 403. If you can provide any socket client example. 

URL: ws://127.0.0.1:5000/socket.io

**Reply by miguelgrinberg** [reply]
The example includes a client. I repeat:

> Use your browser to connect to http://localhost:5000 with the example code I uploaded to see a working solution.

**Comment by osangu** [comment]
Many times passed, But i hope this solution fit for you.

https://github.com/pyropy/fastapi-socketio/issues/51

**Reply by platonh** [reply]
It helped, thank you
